tasks/main.yml

```diff
@@ -28,12 +28,32 @@
   register: containerd_config_default
   when: containerd_config_default_write
 
-- name: Write defaults to config.toml.
+- name: Prepare containerd/config.toml from default config
   copy:
-    dest: /etc/containerd/config.toml
+    dest: /tmp/containerd_config.toml
     content: "{{ containerd_config_default.stdout }}"
+  when: containerd_config_default_write
```

Comment:
Between this line and the new 'Cleanup temporary file' task below; won't this break the role's idempotence? Is there a way we can either just leave the temp file in place, or do this some other way?

Reply:
I am not sure what you see as a problem here. However, let's wait to see if we can define this in a subconfig. imho that would be the best case. 

I am waiting for https://github.com/containerd/containerd/discussions/5413#discussioncomment-650916 here.

Reply:
Could you use a regex filter on the output of the `containerd_config_default.stdout` to add the line needed, and then use that as the content for the write? then it should remain idempotent.

Reply:
I see, it's just because of the idempotent **output**. I thought it was because of the idempotent **function**, that's why I didn't get it. ;)

Thanks for explaining this!